### DISCUSSION TITLE
New substructure highlighting

### DISCUSSION THREAD

**Discussion by c-feldmann** [rdkit/rdkit#4607] [Show and tell] [answered]
Hey,
I created a (hopefully) new Highlighting for RDKit:
![output_15_0](https://user-images.githubusercontent.com/34950800/137324712-d77d5663-584a-4ce1-afb1-24a3cd01f59a.png)
I wanted to ask if you could give me some feedback on the code and if it is worth it to be added to RDKit:
https://github.com/c-feldmann/lassohighlight

**Comment by greglandrum** [comment]
I haven't really looked at the code yet, but I very much like the style of rendering and think you've done a great job of documenting it.

One quick suggestion: it would be a good idea to add a license to that repository and to put a copyright notice in the source files. This is a good general practice in github.

As far as the RDKit is concerned: I think this would be a really good addition. We need to discuss the best way to do that. Ideally we'd have a C++ implementation, but if you aren't a C++ developer we can probably figure out a solution there.

**Reply by c-feldmann** [reply]
Thank you for feedback. I'll implement everything as soon as possible.

The code is written in python and my skills in C++ are not that advanced. However I'll try to translate the functions and see how far I come.

**Reply by greglandrum** [reply]
@c-feldmann : unless you're really motivated to do that, I would suggest holding off a bit. Ignoring the fact that it's in C++, the drawing code is "non-trivial" to work with. I think we can find someone (possibly me) to do the C++ port of your code, but it'll need to wait until after the UGM is done and the 2021.09.1 release is out.

**Comment by DavidACosgrove** [comment]
Hi @c-feldmann, that does look nice. I toyed with the idea of that sort of
highlighting whilst doing the others, but ran out of steam. Many thanks for
making the effort. I agree with @greglandrum that if you’re unfamiliar with
C++ the drawing code is quite a deep deep end to start in. Greg, I’m a bit
tied up at  the moment but should be free to have a look at this in a few
weeks if you don’t fancy it yourself. And I haven’t forgotten that I owe
you a bug fix for the label drawing.
Best,
Dave


On Fri, 15 Oct 2021 at 04:08, Greg Landrum ***@***.***> wrote:

> @c-feldmann <https://github.com/c-feldmann> : unless you're really
> motivated to do that, I would suggest holding off a bit. Ignoring the fact
> that it's in C++, the drawing code is "non-trivial" to work with. I think
> we can find someone (possibly me) to do the C++ port of your code, but
> it'll need to wait until after the UGM is done and the 2021.09.1 release is
> out.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/4607#discussioncomment-1480298>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FTLOKZRWOJOSWCSGSLUG6LMHANCNFSM5F7UQZ5A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by c-feldmann** [reply]
@greglandrum That would be awesome! I'll try to conceptualize the code in some slides so it is easier to port it.
@DavidACosgrove I think at the UGM 2019 the new drawing code was presented and with it some concepts which were not implemented, such as the  lasso-highlight. Since I really liked the idea I tried to implement it myself.

**Reply by c-feldmann** [reply]
@greglandrum & @DavidACosgrove:
I created the [slides](https://github.com/c-feldmann/lassohighlight/blob/master/LassoHighlightingConcept.pdf) explaining the concept of the code. It was a bit rushed so if you need more explanations please let me know.

**Comment by DavidACosgrove** [comment]
Hi Christian,
A somewhat belated thanks for the slides.  Just to let you know that having
won the arm-wrestle with Greg I'm aiming to implement this in time for the
spring release of the toolkit.  I'll keep you updated as to progress.

Best,
Dave


On Wed, Oct 27, 2021 at 4:23 PM Christian W. Feldmann <
***@***.***> wrote:

> @greglandrum <https://github.com/greglandrum> & @DavidACosgrove
> <https://github.com/DavidACosgrove>:
> I created the slides
> <https://github.com/c-feldmann/lassohighlight/blob/master/LassoHighlightingConcept.pdf>
> explaining the concept of the code. It was a bit rushed so if you need more
> explanations please let me know.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/4607#discussioncomment-1546324>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FT7UGKEIA5YFG3X4LTUJARPZANCNFSM5F7UQZ5A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by c-feldmann** [reply]
Thank you very much!

**Comment by DavidACosgrove** [comment, marked as answer]
@c-feldmann.  It's taken 2.5 years, but this is finally implemented as PR #6653.
I am very sorry it has taken so long.  I hope you think it was worth the wait.

**Reply by c-feldmann** [reply]
Thank you so much for taking the time to implement my suggestion. I really think it was worth the time! :)
